questions.ts

```diff
@@ -629,52 +633,88 @@ export const askQuestions = async (
           break;
         }
-        if (selectedSource === "exampleFile") {
-          program.dataSources.push(EXAMPLE_FILE);
-        } else if (selectedSource === "file" || selectedSource === "folder") {
-          // Select local data source
-          const selectedPaths = await selectLocalContextData(selectedSource);
-          for (const p of selectedPaths) {
+        switch (selectedSource) {
+          case "exampleFile": {
+            program.dataSources.push(EXAMPLE_FILE);
+            break;
+          }
+          case "file":
+          case "folder": {
+            const selectedPaths = await selectLocalContextData(selectedSource);
+            for (const p of selectedPaths) {
+              program.dataSources.push({
+                type: "file",
+                config: {
+                  path: p,
+                },
+              });
+            }
+            break;
+          }
+          case "web": {
+            const { baseUrl } = await prompts(
+              {
+                type: "text",
+                name: "baseUrl",
+                message: "Please provide base URL of the website: ",
+                initial: "https://www.llamaindex.ai",
+                validate: (value: string) => {
+                  if (!value.includes("://")) {
+                    value = `https://${value}`;
+                  }
+                  const urlObj = new URL(value);
+                  if (
+                    urlObj.protocol !== "https:" &&
+                    urlObj.protocol !== "http:"
+                  ) {
+                    return `URL=${value} has invalid protocol, only allow http or https`;
+                  }
+                  return true;
+                },
+              },
+              handlers,
+            );
+
             program.dataSources.push({
-              type: "file",
+              type: "web",
               config: {
-                path: p,
+                baseUrl,
+                prefix: baseUrl,
+                depth: 1,
               },
             });
+            break;
           }
-        } else if (selectedSource === "web") {
-          // Selected web data source
-          const { baseUrl } = await prompts(
-            {
-              type: "text",
-              name: "baseUrl",
-              message: "Please provide base URL of the website: ",
-              initial: "https://www.llamaindex.ai",
-              validate: (value: string) => {
-                if (!value.includes("://")) {
-                  value = `https://${value}`;
-                }
-                const urlObj = new URL(value);
-                if (
-                  urlObj.protocol !== "https:" &&
-                  urlObj.protocol !== "http:"
-                ) {
-                  return `URL=${value} has invalid protocol, only allow http or https`;
-                }
-                return true;
+          case "db": {
```

Comment:
We'll allow multiple database sources, each database source can have multiple queries, but we only ask one query in the CLI, user can reconfigure it later in the loaders.yaml file.

Reply:
that's great. keep the UI simple as possible